packages/manager/apps/web/client/app/domain/zone/zone.routing.js

```diff
@@ -1,13 +1,137 @@
 import controller from './domain-zone-dns.controller';
 import template from './domain-zone-dns.html';
 
+const commonResolveForZoneHistory = {
+  goToZoneHistory: /* @ngInject */ ($state) => (params) =>
+    $state.go('app.domain.product.zone-history', params),
+  breadcrumb: /* @ngInject */ ($translate) =>
+    $translate.instant('zone_history'),
+};
+const commonResolve = {
+  goToZoneHistory: /* @ngInject */ ($state) => (params) =>
+    $state.go('app.domain.product.zone-history', params),
+  breadcrumb: /* @ngInject */ ($translate) => $translate.instant('domain_zone'),
+};
+
 export default /* @ngInject */ ($stateProvider) => {
-  const commonResolve = {
-    goToZoneHistory: /* @ngInject */ ($state) => (params) =>
-      $state.go('app.zone.details.zone-history', params),
-    breadcrumb: /* @ngInject */ ($translate) =>
-      $translate.instant('domain_zone'),
-  };
+  $stateProvider.state('app.domain.product.zone-history', {
+    url: '/zone-history',
+    views: {
+      domainView: {
+        component: 'domainZoneDashboardHistory',
+      },
+    },
+    resolve: {
+      ...commonResolveForZoneHistory,
+      goToDnsData: /* @ngInject */ ($state) => (url) =>
+        $state.go('app.domain.product.zone-history.view', { url }),
+      goToDnsRestore: /* @ngInject */ ($state) => (url, chosenDate) =>
+        $state.go('app.domain.product.zone-history.restore', {
+          url,
+          chosenDate,
+        }),
+      goBack: /* @ngInject */ ($state, $stateParams) => () =>
+        $state.go('app.domain.product.zone', $stateParams),
+      goToDiffViewer: /* @ngInject */ ($state) => (
+        dnsEntriesForComparison,
+        zoneName,
+      ) =>
+        $state.go('app.domain.product.diff-tool-viewer', {
+          selectedDates: dnsEntriesForComparison
+            .filter((u) => u.active)
+            .map((u) => u.date),
+          productId: zoneName,
+        }),
+    },
+    params: {
+      productId: null,
+    },
+  });
+
+  $stateProvider.state('app.domain.product.diff-tool-viewer', {
+    url: '/diff-tool-viewer',
+    views: {
+      domainView: {
+        component: 'domainZoneDiffToolViewerHistory',
+      },
+    },
+    resolve: {
+      ...commonResolveForZoneHistory,
+      goBack: /* @ngInject */ ($state, $stateParams) => () =>
+        $state.go('app.domain.product.zone-history', $stateParams),
+    },
+    params: {
+      selectedDates: null,
+      productId: null,
+    },
+  });
+
+  $stateProvider.state('app.domain.product.zone-history.view', {
+    url: '/view',
+    layout: { name: 'modal', keyboard: true },
+    views: {
+      modal: {
+        component: 'zoneHistoryView',
+      },
+    },
+    params: {
+      url: {
+        type: 'string',
+        value: null,
+      },
+    },
+    resolve: {
+      ...commonResolveForZoneHistory,
+      url: /* @ngInject */ ($stateParams) => $stateParams.url,
```

Comment:
**question:** What does `url` indicate here? It is confusing for me.

Reply:
This is the url given by the response of operation/domain

Reply:
Ah ok found it this is the zoneFileUrl used to retrieve the file data:
![image](https://github.com/ovh/manager/assets/8767496/3f6bb586-0173-4c94-9335-c5008a0b8470)

Reply:
I added a description here then:

![image](https://github.com/ovh/manager/assets/8767496/f6f51990-b917-47a1-be3f-25f23172c35f)